### ISSUE TITLE
EOS errors from ideal ions at high density / low temperature

### ISSUE THREAD

**Issue by evbauer** [MESAHub/mesa#565]
See discussion on c58b3fc7cae532008ef437231033dd86718764b2, which was reverted.

The ideal ion free energy in Skye gives negative entropy at sufficiently high density. This is a problem not just for Skye, but also for ideal, which is meant to be our backstop EOS. Therefore we end up having no EOS coverage at all in the region labeled 'nothing' in this plot:

![image](https://github.com/MESAHub/mesa/assets/18405113/bcfb91ed-b468-433f-a663-128d9111f356)

The proper physical solution is to account for ion degeneracy, but that might require some tabulations of fermi-dirac integrals?

**Comment by evbauer** [comment]
Practically, I'm most concerned about falling off Skye near logT = 10 on this plot. Massive WDs could possibly get there, and falling from Skye to HELM here would likely cause problems.

I'm also wondering if this unphysical ideal ion term may have had an unintended impact on the discussion of quantum melting in Appendix D of the [Skye paper](https://ui.adsabs.harvard.edu/abs/2021ApJ...913...72J/abstract). See figure 18 in the paper. The shape of the quantum melting region for C/O looks suspiciously similar to the shape of the region where Skye is failing and falling back to HELM/nothing as it approaches logT=10. (The plot above is for a C/O mixture)

**Comment by fxt44** [comment]
i have a helm-style table that treats ions as fermions (an approximation). in the low-density limit it is ideal gas and at larger densities captures the onset of ion degeneracy, where ideal gas expressions fail. the entropy remains positive and smooth. the cost is additional table machinery. i also haven't explicitly tested it in a few years.

**Comment by Debraheem** [comment]
@evbauer , i think this issue was addressed in https://github.com/MESAHub/mesa/issues/589? can i close it?

**Comment by evbauer** [comment]
I don't think so. #589 addressed the upper-left corner of the rho-T plot. This is an issue on the lower-right corner, and I believe it persists until we do something like @fxt44 suggested.

**Comment by Debraheem** [comment]
> I don't think so. [#589](https://github.com/MESAHub/mesa/issues/589) addressed the upper-left corner of the rho-T plot. This is an issue on the lower-right corner, and I believe it persists until we do something like [@fxt44](https://github.com/fxt44) suggested.

ahh yes, thanks for correcting me!
